AssetTracking/AssetTracking/Helpers/GraphService.cs

```diff
@@ -0,0 +1,189 @@
+                var pictureMemoryStream = new MemoryStream();
+                await pictureStream.CopyToAsync(pictureMemoryStream);
+
+                // Convert stream to byte array.
+                var pictureByteArray = pictureMemoryStream.ToArray();
+
+                // Convert byte array to base64 string.
+                var pictureBase64 = Convert.ToBase64String(pictureByteArray);
+
+                return "data:image/jpeg;base64," + pictureBase64;
+            }
+            catch (Exception e)
+            {
+                switch (e.Message)
+                {
+                    case "ResourceNotFound":
+                        // If picture not found, return the default image.
+                        return "data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4NCjwhRE9DVFlQRSBzdmcgIFBVQkxJQyAnLS8vVzNDLy9EVEQgU1ZHIDEuMS8vRU4nICAnaHR0cDovL3d3dy53My5vcmcvR3JhcGhpY3MvU1ZHLzEuMS9EVEQvc3ZnMTEuZHRkJz4NCjxzdmcgd2lkdGg9IjQwMXB4IiBoZWlnaHQ9IjQwMXB4IiBlbmFibGUtYmFja2dyb3VuZD0ibmV3IDMxMi44MDkgMCA0MDEgNDAxIiB2ZXJzaW9uPSIxLjEiIHZpZXdCb3g9IjMxMi44MDkgMCA0MDEgNDAxIiB4bWw6c3BhY2U9InByZXNlcnZlIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPg0KPGcgdHJhbnNmb3JtPSJtYXRyaXgoMS4yMjMgMCAwIDEuMjIzIC00NjcuNSAtODQzLjQ0KSI+DQoJPHJlY3QgeD0iNjAxLjQ1IiB5PSI2NTMuMDciIHdpZHRoPSI0MDEiIGhlaWdodD0iNDAxIiBmaWxsPSIjRTRFNkU3Ii8+DQoJPHBhdGggZD0ibTgwMi4zOCA5MDguMDhjLTg0LjUxNSAwLTE1My41MiA0OC4xODUtMTU3LjM4IDEwOC42MmgzMTQuNzljLTMuODctNjAuNDQtNzIuOS0xMDguNjItMTU3LjQxLTEwOC42MnoiIGZpbGw9IiNBRUI0QjciLz4NCgk8cGF0aCBkPSJtODgxLjM3IDgxOC44NmMwIDQ2Ljc0Ni0zNS4xMDYgODQuNjQxLTc4LjQxIDg0LjY0MXMtNzguNDEtMzcuODk1LTc4LjQxLTg0LjY0MSAzNS4xMDYtODQuNjQxIDc4LjQxLTg0LjY0MWM0My4zMSAwIDc4LjQxIDM3LjkgNzguNDEgODQuNjR6IiBmaWxsPSIjQUVCNEI3Ii8+DQo8L2c+DQo8L3N2Zz4NCg==";
+                    case "EmailIsNull":
+                        return JsonConvert.SerializeObject(new { Message = "Email address cannot be null." }, Formatting.Indented);
+                    default:
+                        return null;
+                }
+            }
+        }
+        public static async Task<Stream> GetPictureStream(GraphServiceClient graphClient, string email, HttpContext httpContext)
+        {
+            if (email == null) throw new Exception("EmailIsNull");
+
+            Stream pictureStream = null;
+            try
+            {
+                try
+                {
+                    // Load user's profile picture.
+                    pictureStream = await graphClient.Users[email].Photo.Content.Request().GetAsync();
+                }
+                catch (ServiceException e)
+                {
+                    if (e.Error.Code == "GetUserPhoto") // User is using MSA, we need to use beta endpoint
+                    {
+                        // Set Microsoft Graph endpoint to beta, to be able to get profile picture for MSAs 
+                        graphClient.BaseUrl = "https://graph.microsoft.com/beta";
```

Comment:
Consider using the `GraphResourceId` property here: 
`graphClient.BaseUrl = configuration.GraphResourceId + beta;`
[https://github.com/microsoftgraph/aspnet-assettracking-sample/blob/ea4e13c937fa32f3eff082dbc2ce53b1cf0d92d3/AssetTracking/AssetTracking/Extensions/AzureAdOptions.cs#L13](url)

You can also have the `beta` and `v1.0` values as constants so that they are defined once and re-used throughout this class.